libr/util/big.c

```diff
@@ -170,6 +168,9 @@ R_API char *r_big_to_hexstr(RNumBig *b) {
 
 	size = 3 + 2 * R_BIG_WORD_SIZE * (j + 1) + ((b->sign > 0)? 0: 1);
 	char *ret_str = calloc (size, sizeof (char));
```

Comment:
sizeof(char) is always 1, i dont think we need this indirection. also this shuold be using malloc, because the whole buffer is later written, so no UB can happen

Reply:
define and assign size in the right place

Reply:
"0x0" is not an hex string

Reply:
> sizeof(char) is always 1, i dont think we need this indirection. also this shuold be using malloc, because the whole buffer is later written, so no UB can happen

No, in testing, `malloc` return a buffer already be written with a very long string. But the leading zero is omitted, so the buffer is not fully written in `r_big_to_hexstr`. In the end, there's some unexpected chars after actual hexstring.


> "0x0" is not an hex string

```python
hex(0) -> '0x0'
```